source/extensions/common/wasm/context.cc

```diff
@@ -994,7 +994,16 @@ WasmResult Context::grpcCall(absl::string_view grpc_service, absl::string_view s
                              uint32_t* token_ptr) {
   GrpcService service_proto;
   if (!service_proto.ParseFromArray(grpc_service.data(), grpc_service.size())) {
-    return WasmResult::ParseFailure;
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.wasm_cluster_name_envoy_grpc")) {
+      auto cluster_name = std::string(grpc_service.substr(0, grpc_service.size()));
+      const auto thread_local_cluster = clusterManager().getThreadLocalCluster(cluster_name);
+      if (thread_local_cluster == nullptr) {
+        return WasmResult::BadArgument;
```

Comment:
I think we should return `WasmResult::ParseFailure` here, since previously neither `grpcCall` nor `grpcStream` could return `WasmResult::BadArgument`, so this would result in a new code path for callers, and since we're not bumping the ABI version, I don't think that we should do that.

Having said that, this is a "better" result code, so feel free to add a `TODO` saying that we should switch to `WasmResult::BadArgument` in the next iteration of the ABI.

Reply:
done